src/Arcus.Security.Providers.AzureKeyVault/Authentication/ManagedServiceIdentityAuthenticator.cs

```diff
@@ -1,19 +1,20 @@
 ﻿using System.Threading.Tasks;
+using Arcus.Security.Providers.AzureKeyVault.Authentication.Interfaces;
 using Microsoft.Azure.KeyVault;
 using Microsoft.Azure.Services.AppAuthentication;
 
-namespace Arcus.Security.Providers.AzureKeyVault.Factories
+namespace Arcus.Security.Providers.AzureKeyVault.Authentication
 {
     /// <summary>
-    /// <see cref="KeyVaultClientFactory"/> implementation using Azure Managed Service Identity
+    ///     Azure Key Vault authentication by using Azure Managed Service Identity
     /// </summary>
-    public class MsiKeyVaultClientFactory : KeyVaultClientFactory
+    public class ManagedServiceIdentityAuthenticator : IKeyVaultAuthenticator
```

Comment:
Not sure if this is already published but otherwise this would be a breaking change and we should use obsolete instead? But I think I'm wrong and this isn't published yet